### DISCUSSION TITLE
Найти треки по ключевому слову в своей библиотеке

### DISCUSSION THREAD

**Discussion by B1gV** [Chimildic/goofy#48] [Помощь с алгоритмом] [answered]
Приветствую!

Хочу сделать плейлист со всеми акустическими версиями треков исполнителей из моей библиотеки. Основная идея в том, что у таких треков в названии почти всегда присутствует слово "acoustic". Мой код:

```
function ThemePlaylist () {

let tracks = Source.getArtistsTracks({
    artist: {
        followed_include: true,
    },
    album: {
        release_date: { sinceDays: 1000, beforeDays: 0},
    },
});

Filter.match (tracks, 'acoustic');
```
У него 2 проблемы:

1. Невозможно получить все треки исполнителей из подписок, нужен какой-то фильтр. Ничего лучше, чем фильтровать по дате, менять дату и снова запускать код я не придумал. Но это еще полбеды.
2. Куда большая проблема - Filter.match почему то фильтрует треки у которых искомое выражение присутствует и в названии трека, и в названии альбома одновременно. Это, понятно, дает только небольшую часть необходимых треков. Так, что даже по частям получается далеко не все, что нужно.

**Comment by Chimildic** [comment, marked as answer]
1. Действительно сложно получить все треки отслеживаемых исполнителей. В процессе совершается множество дробленных запросов. Spotify заставляет делать паузы из-за наплыва однотипных запросов в короткий промежуток времени. 
2. Нашел логическую ошибку в условии отбора в функциях `match*`. Обновите код, должно помочь.

**Reply by B1gV** [reply]
Теперь match работает как ИЛИ, спасибо!
